### ISSUE TITLE
How much time / money would it take / will you give to have this published to a p2 site?

### ISSUE THREAD

**Issue by nedtwigg** [wjywbs/javacef#39]
There are lots of people who want this, but having everyone build their own native binaries is inefficient.  Most of the issues are about building it, not using it or reporting actionable bugs that could improve the software.

What would it take to get this project stable, with published binaries?  Possibly the eclipse foundation could do this and make it an optional distribution alongside SWT.

@wjywbs could you give a schedule estimate?  Cost estimate?  Can anyone estimate what they could pitch in, financially or with engineering time, to contribute to this?

I am fairly skilled with gradle, and it is a good tool for tying together disparate command line tools in a repeatable way.  I've also written a [gradle plugin](https://github.com/diffplug/goomph) which can handle p2 repositories.

Maybe @wjywbs and I could sit down and setup a CI build that publishes everything to a public amazon bucket.  I'm trying to convince the eclipse foundation to take this project on, but if we have something working that might make it easier to convince them.

**Comment by wjywbs** [comment]
The base CEF version (3.1750) hasn't been updated for a long time. 3.1750 is stable, but I'm not sure if it's a good idea to put this project in some repo without enough maintenance in the future. Updating to later version (2526) was difficult due to changes in finding resource paths. Hopefully I would have time to try the current latest version again.

This is my personal project. I aimed to release once a year, but I didn't have enough time to achieve this. Considering that CEF upstream releases every 2 months, things get outdated quickly.

I had a bintray account for binary releases, and it shouldn't take too much time to write packaging scripts for 3.1750. Contributions are also welcome.

**Comment by nedtwigg** [comment]
> I'm not sure if it's a good idea to put this project in some repo without enough maintenance in the future

I agree!  You've built something really valuable, but I think it's not getting used to its potential because
1. A lot of people who want it don't have the skills to build their own binaries
2. People are worried that there isn't a team around it to keep it up-to-date

I'd like to help you grow your project to reach a bigger audience.  If your project became the seed of "swt-chromium", hosted on eclipse.org with their build infrastructure, would that make you happy or sad?

> 3.1750 hasn't been updated for a long time

For anyone else who didn't know, 3.1750 was released June 2014, and supports Chromium 33.  Current Chromium is 54.  Some notable improvements between 33 and 54:
- 35: shadow DOM (I believe critical to newer js frameworks like react)
- 39: 64-bit mac support (critical because SWT is now 64-bit only on mac)
- 40 & 41: 113 security issues

2526 which you've started would bring us to Chromium 47.  Looks like latest stable chromium is 2704 which has Chromium 51.

[CEF Builds](https://cefbuilds.com/)
[Chromium changelog](https://en.wikipedia.org/wiki/Chromium_%28web_browser%29#2014)

> This is my personal project.

Would you consider making it a semi-professional personal project?  I know lots of SWT users who really want this.  How much money would we need to raise for you find the time to get it updated?  If possible, I'd love to chat in realtime, my contact info is in my profile.

**Comment by nedtwigg** [comment]
It looks like @inaauto has made some progress on 3.2526 as well:
https://github.com/inaauto/javacef/commits/master

**Comment by nedtwigg** [comment]
Just wanted to let you know that Texas Instruments has open sourced their Chromium SWT embed, and if you'd like you can take a look [here](https://github.com/diffplug/swt-chromium/tree/ti_contribution).

I've done my best to compare your work to TI's [here](https://github.com/diffplug/swt-chromium/issues/2#issuecomment-262389342).  It looks like your project has the best build system and compatibility with more modern versions of CEF3, and TI's implementation has some features that JavaCEF doesn't have yet (e.g. execute java and javascript in the browser).

So that's how far along I am.  I'd love your help and/or comments if you've got time, but of course I understand if you're busy :)

**Comment by nedtwigg** [comment]
Just FYI, the eclipse foundation has raised money to build an officially-supported SWT Browser based on embedded Chromium.  It is [taking bids](https://projects.eclipse.org/development_effort/implement-swtchromium-integration) on the project from qualified developers (Jingyi is as-qualified as possible for this effort, I believe).  If Jingyi is not able to find time, however, anyone can bid to complete the project, or to make a [directed donation](http://www.eclipse.org/contribute/dev_program.php) to the eclipse foundation which will go towards reviving the SWT Browser with a modern embedded Chromium.  It's called FEEP (Friends of Eclipse Enhancement Program).

Bids for this cycle are required by December 23, 2016.
